### DISCUSSION TITLE
Add Trusted Types support

### DISCUSSION THREAD

**Discussion by timneutkens** [vercel/next.js#62618] [Ideas]
### Describe the feature you'd like to request

Backstory and initial implementation can be found here: #13509

### Describe the solution you'd like

As mentioned in #13509 Trusted Types helps prevent XSS attack, we're planning to add support for it in Next.js natively so that there is a default configuration.

Someone from the Chrome team is going to start working on adding Trusted Types to Next.js early January.

**Comment by jgoping** [comment]
The pull request above is to add a tool called [tsec](https://github.com/google/tsec) to the linting process to check for Trusted Types violations. Going forward, I plan to create more pull requests to fix all of the violations detected by tsec. Also, tsec is not guaranteed to catch all violations, so I plan to find and fix violations that occur when running Next.js applications locally or from running unit/integration tests. After all of these violations are fixed, then application developers can choose to enforce Trusted Types without being blocked by the framework.

**Comment by manuel-mauky** [comment]
Any news on this?

**Comment by samcx** [comment]
Moving this to <picture data-single-emoji=":nextjs:" title=":nextjs:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/nextjs/2a85d633ae594556.png" alt=":nextjs:" align="absmiddle"></picture> discussions since this is a feature request.

**Comment by alexaka1** [comment]
Afaik React already supports Trusted Types: https://github.com/facebook/react/pull/16157/files

**Comment by alexaka1** [comment]
Hmm, so does Next?
https://github.com/vercel/next.js/pull/13509
No never mind, this was closed in favor of this issue, which was closed and demoted to a discussion. Catch 22.
